icevision/models/torchvision_models/faster_rcnn/show_results.py

```diff
@@ -31,3 +36,103 @@ def show_results(
         denormalize_fn=denormalize_fn,
         show=show,
     )
+
+
+def get_losses(
+    model: nn.Module,
+    dataset: Dataset,
+) -> Tuple[List[dict], dict]:
+    """
+    Gets a dataset and a model as input and returns losses calculated per image + losses stats.
+    In case of faster_rcnn those are loss_classifier, loss_box_reg and loss_total
+    (sum of the first 2 plus RPN related losses, e.g. sum of 4 terms).
+
+    Arguments
+    ---------
+    model: nn.Module
+    dataset: Dataset
+
+    Returns
+    -------
+    samples_plus_losses: those are the samples (dictionaries) contained in the original dataset
+                          enriched with losses.
+    losses_stats: dictionary containing losses stats (min, max, mean, 1-25-50-75-99 quantiles)
+                  for each one of the losses.
+    """
+    device = torch.device("cpu")
```

Comment:
Can we also run this on GPU?

Reply:
Given I am running the forward image by image, so `batch_size=1` (approach which cannot be changed) I thought there would be no benefit in moving the samples to GPU.
Just overheard for nothing gained. No?

Reply:
I think even with `bs=1` it might still be faster on GPU, we can try and see =)

Reply:
I will then arrange this and move the tensors to whatever device the model is on.

Reply:
Done. If the model is on GPU then the tensors will be moved to CUDA too.